File: backend/src/main/java/com/hcmunre/library/entity/ThongBao.java
package com.hcmunre.library.entity;

import com.hcmunre.library.enums.LoaiThongBao;
import jakarta.persistence.*;
import lombok.*;

import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
public class ThongBao extends BaseEntity{
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID maThongBao;

    @Column(nullable = false)
    private String tieuDe;

    @Column(nullable = false)
    private String noiDung;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private LoaiThongBao loaiThongBao;

    private boolean daDoc;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ma_nguoi_dung", nullable = false)
    private NguoiDung nguoiDung;
}

File: frontend/src/types/thongbao.ts
export type LoaiThongBao = 'HE_THONG' | 'NHAC_NHO' | 'CANH_BAO' | 'DAT_CHO'

export interface ThongBao {
  maThongBao: string
  tieuDe: string
  noiDung: string
  loaiThongBao: LoaiThongBao
  daDoc: boolean
  ngayTao: string
  maNguoiDung?: string
}

